src/query/storages/system/src/columns_table.rs

```diff
@@ -142,15 +142,42 @@ impl ColumnsTable {
         &self,
         ctx: Arc<dyn TableContext>,
         push_downs: Option<PushDownInfo>,
-    ) -> Result<Vec<(String, String, TableField)>> {
+    ) -> Result<Vec<(String, String, String, TableField)>> {
         let database_and_tables = dump_tables(&ctx, push_downs).await?;
 
-        let mut rows: Vec<(String, String, TableField)> = vec![];
+        let mut rows: Vec<(String, String, String, TableField)> = vec![];
         for (database, tables) in database_and_tables {
             for table in tables {
-                let fields = generate_fields(&ctx, &table).await?;
-                for field in fields {
-                    rows.push((database.clone(), table.name().into(), field.clone()))
+                if table.engine() != VIEW_ENGINE {
+                    let schema = table.schema();
+                    let field_comments = table.field_comments();
+                    let n_fields = schema.fields().len();
+                    for (idx, field) in schema.fields().iter().enumerate() {
+                        // compatibility: creating table in the old planner will not have `fields_comments`
+                        let comment = if field_comments.len() == n_fields
+                            && !field_comments[idx].is_empty()
+                        {
+                            format!("'{}'", &field_comments[idx].as_str().replace('\'', "\\'"))
```

Comment:
how about use `format!("'{:?}'", &field_comments[idx])`?

Reply:
> how about use `format!("'{:?}'", &field_comments[idx])`?

O I got you.

Reply:
Note:  can not use debug print, will add double quote